runtime/metricsview/query.go

```diff
@@ -23,6 +23,11 @@ type Query struct {
 	Offset              *int64      `mapstructure:"offset"`
 	TimeZone            string      `mapstructure:"time_zone"`
 	UseDisplayNames     bool        `mapstructure:"use_display_names"`
+
+	// internal state
+	inlineBaseSelect bool          `mapstructure:"-"`
+	inlineDims       map[any][]any `mapstructure:"-"` // dimension -> column values
+	inlineMeasures   map[any][]any `mapstructure:"-"` // measure -> column values
```

Comment:
It somehow feels dirty to add this internal state to the `Query` type (it's widely used many places, but the internal state here applies to a specific internal optimization). Contrast this with the `rewriteQueryDruidExactify`, which despite rewriting the query does not need to add any internal fields. To the extent possible, it would be nice if the two-phase approach could live mainly in the `executor_rewrite_two_phase` file, and only need minimal changes to the generic types in this package. 

Did you consider implementing the optimization by traversing the `AST` type instead of `Query`, similar to how `rewriteApproxComparisons` works?

Reply:
removed